backend/src/app.js

```diff
@@ -341,6 +343,104 @@ app.get("/admin/review", async (req, res) => {
   }
 });
 
+app.get("/library", async (req, res) => {
+  try {
+    console.log("Received request at /book:", req.query);
+    let { search } = req.query;
+
+    const db = DbService.getDb();
+
+    if (!search) {
+      return res.status(400).send("Missing required fields.");
+    }
+
+    search = `%${search}%`;
+    console.log("search:", search);
+    const books = await db
+      .select()
+      .from(schema.book)
+      .leftJoin(schema.writtenby, eq(schema.book.isbn, schema.writtenby.isbn))
+      .leftJoin(schema.author, eq(schema.writtenby.author, schema.author.id))
+      .where(
+        or(
+          like(schema.book.title, search),
+          like(schema.author.name, search),
+          like(schema.book.isbn, search),
+        ),
+      );
+    if (!books) {
+      return res
+        .status(400)
+        .send(
+          ErrorService.handleError(
+            `"${search}" doesn't exist in this library.`,
+          ),
+        );
+    }
+
+    res.status(200).json(books);
+  } catch (err) {
+    if (res.closed) {
+      return;
+    }
+    return res
+      .status(err.status || 500)
+      .send(
+        ErrorService.handleError(
+          err,
+          "An error occurred while fetching books.",
+        ),
+      );
+  }
+});
+
+app.get("/book", async (req, res) => {
+  try {
+    console.log("Received request at /book:", req.query);
+    let { search } = req.query;
+
+    const db = DbService.getDb();
+
+    if (!search) {
+      return res.status(400).send("Missing required fields.");
+    }
+
+    search = `%${search}%`;
+    console.log("search:", search);
+    const book = await db
+      .select()
+      .from(schema.book)
+      .leftJoin(schema.writtenby, eq(schema.book.isbn, schema.writtenby.isbn))
+      .leftJoin(schema.author, eq(schema.writtenby.author, schema.author.id))
+      .where(like(schema.book.isbn, search));
```

Comment:
We can just check equality here 
```suggestion
      .where(eq(schema.book.isbn, isbn));
```